### ISSUE TITLE
Citable repository to reproduce the numerical results

### ISSUE THREAD

**Issue by ranocha** [ketch/positiveRK#12]
(With DOI etc.)

**Comment by snuesslein** [comment]
The Code for the Plots is in the `latex/plots/` sub-directory. But i have too change some parts in the underlying Python Code in the main directory before putting it in citable repository.

**Comment by ketch** [comment]
@snuesslein This will be the very last step once we are ready to submit the paper.  No need to spend time on it until then, since we may change what examples are included or how we present them.

**Comment by ranocha** [comment]
Sorry, I should have written what @ketch wrote above.

**Comment by ketch** [comment]
I've created a repository, copying over the files that @ranocha suggested (and one more that was necessary):

https://github.com/ketch/Positive_RK_Reproducibility

I tried running the first notebook (Convergence.ipynb) but I get several errors, starting with an index error at cell 17.  Does this code run without errors for you guys?

**Comment by ranocha** [comment]
Yes, it works for me. Do you have Mosek installed?

I will draft a README.md with installation instructions and make a PR.

**Comment by ketch** [comment]
No.  Can you write a README that indicates what needs to be installed?

**Comment by ketch** [comment]
After installing Mosek, I still get the same index error.  Please clone the reproducibility repo and see if the notebook runs for you.

**Comment by ranocha** [comment]
I cloned the repo and ran `Convergence.ipynb`. I don't get any errors.

Which errors do you get?

**Comment by ranocha** [comment]
@ketch Could you please draft a new release of nodepy? We use a feature that has been introduced after the latest release and it is much easier to tell users to just `pip install nodepy` than to clone the repository.

xref https://github.com/ketch/nodepy/issues/65

**Comment by ranocha** [comment]
@ketch Did you also obtain and install a (free academic) license of Mosek?

**Comment by ketch** [comment]
> @ketch Did you also obtain and install a (free academic) license of Mosek?

No; can you add instructions (or a link to instructions) to the README?

**Comment by ketch** [comment]
>Which errors do you get?

Here's the first one, from cell 7.  `u_ref` is 100x1.

![Screen Shot 2020-05-10 at 1 37 42 PM](https://user-images.githubusercontent.com/697763/81496905-b1f31f80-92c3-11ea-8d90-e5c319572638.png)

**Comment by ranocha** [comment]
I've added installation instructions to the Readme in the reproducibility repository. If these work for you, we can copy them to this repository.

**Comment by ketch** [comment]
Thanks; that eliminated the error I was getting.  I'm a little confused by the fact that I didn't just get an import error, and grepping for "mosek" in the whole directory returns nothing.

Anyway, I'm running all the notebooks now.

**Comment by ketch** [comment]
I've been running `Advection_dffusion_reaction.ipynb` for a couple of hours and it hasn't finished.  I guess this is expected?  If the most costly part is the computation of a reference solution, perhaps we should save that to a file to avoid recomputing it.

**Comment by ranocha** [comment]
I think I might have missed that cell while cleaning up the notebooks. One of the first few cells takes really long but isn't needed and should be removed. You should be able to interrupt the kernel and run all other cells to get the plots.

**Comment by ketch** [comment]
Okay, I finished running everything and I got all the figures in the paper.  Only one figure looks noticeably different; I'm attaching it here.  Can you rerun the code and see if you get my version or the one in the paper?
[b_Advection_Decay.pdf](https://github.com/ketch/positiveRK/files/4605821/b_Advection_Decay.pdf)

**Comment by ranocha** [comment]
That's a plot generated by `Convergence.ipynb`, isn't it? Figure 4b in the manuscript?

If I rerun the notebook in the reproducibility repository, I get exactly the version that's in the paper. In particular, I don't get entries for `Order = 2`.

**Comment by ketch** [comment]
What shall we do about this? 

@snuesslein can you run it and post the figure you get?

**Comment by ketch** [comment]
After updating all libraries and restarting the kernel and running, I get the same figure.  Here are my versions.
![Screen Shot 2020-05-11 at 9 25 33 AM](https://user-images.githubusercontent.com/697763/81530383-6c892d80-9369-11ea-8a19-5b86d74d427f.png)

**Comment by ranocha** [comment]
Okay, that's good. I will add the versions of the libraries I used to the readme of the reproducibility repository.

**Comment by snuesslein** [comment]
I get the same result as @ketch . I also tried to reproduce the the plot @ranocha but was not able to do this.

**Comment by ketch** [comment]
In that case I suggest that we use the figure @snuesslein and I get, in the paper.

**Comment by ketch** [comment]
Also: nodepy 0.9 is released, and the reproducibility repo for this paper has a DOI.  See the badge in the README:

https://github.com/ketch/Positive_RK_Reproducibility/blob/master/README.md

**Comment by ranocha** [comment]
Okay. Will you please push the figure to this paper and update the version number in the reproducibility repository?

**Comment by snuesslein** [comment]
![Version](https://user-images.githubusercontent.com/57658753/81545455-33f15000-9379-11ea-9667-742b54c0417a.png)
These are the versions I am using. I think I also got the version of the plot @ranocha is getting at some point. Yesterday I already looked at older versions but could not find it. 
I am also a bit irritated by the fact that we require different orders.

**Comment by snuesslein** [comment]
I think I found one problem with the setup I have chosen. I will make a pull request after I fully investigated it

**Comment by snuesslein** [comment]
The dt i have chosen initially is too large. If we have a look at the plot of the convergence, we can see that the error of the unadapted method increases drastically for $\Delta t \approx 0.017$. This is also the point when we can observe oscillations in the unadapted solution. The dt of the example was $\Delta t = 0.02$ which is obviously larger. 
It was quite a nice example of the fact, that we can suppress oscillation in some cases but this is not the point that we want to make with the plot.

So i changed the dt such that it is stable. Therefore all adaptions are of 4th order, which is also better because the lowest order appearing in the convergence plot is 3.

This does not fully explain why we got different plots, but the change would probably make it easier to reproduce.

I already changed it in the paper and made a pull request to the reproducibility repository.
